README.md

```diff
@@ -0,0 +1,168 @@
+
+So Proving work on the _"workChain"_ has very different properties and will be a separate project of its own. But _this_ module we be able verify _against_ the workChain (who's trust was establised elsewhere)
+
+<!-- ```javascript
+var txHash = '0xb53f752216120e8cbe18783f41c6d960254ad59fac16229d4eaec5f7591319de'
+eP.getTransactionProof(txHash).then((result)=>{
+  // console.log(result) // I now have a proof object
+
+  // I can now verify the proof against a blockhash I trust.
+  var myTrustedBlockHash = Buffer.from('f82990de9b368d810ce4b858c45717737245aa965771565f8a41df4c75acc171','hex')
+  var verified = EP.transaction(result.path, result.value, result.parentNodes, result.header, myTrustedBlockHash)
+  console.log(verified) // true
+}).catch((e)=>{console.log(e)})
+``` -->
+
+
+## Testing
+
+
+The tx and receipt tests use infura right now (because I dont have a completely full node). The account and storage tests point to localhost:8545 because infura doenst yet support the rpc call needed to attain them `eth_getProof` (from EIP 1186).
+
+Its all data currently on Ethereum Mainnet.
+
+These tests hit Infura really hard because every tx or receipt proof requires multiple RPC calls (1 for each tx in the particular block). please be considerate. If you have a full archive node, re-point the rpc calls locally or comment out all but one tests at a time.
+
+Thanks to @simon-jentzsch, for EIP-1186 to make this data available from Geth and Parity clients.
+
+
+## Next Level Shit
+
+
+long term goal is are light clients that can validate an entire state transition. It would need proofs for all data touched during the state transition (tx).
+
+We also would like wallets to display only data that is proven.
+
+To complete the functionality attempted by this tool, a "light-client" tool (that downloads all the hashes and validates the work between the) will have to be built. The output of which will be a "workChain" which can interface with eth-proof to finally begin to leverage some of the really useful security properties of PoW blockchains.
+
+proving absence (or null or undefined):
+This is a really cool feature of the state tree. You can prove a key does not exist in the tree. In fact you need this property for certain aplications. Unfortunately I dont have that working yet, but it's the next feature, but its much cleaner to modify merkle-patricia-tree in order to support it. Check back in a few days.
+
+There will be some major changes to the format of `Branch`. I have realized that a `branch` is really just a sparse `tree`. instead of using the array of nodes, I should be putting those nodes in the tree and then doing a regular `tree.get` to pull it back out. If the tree finds it, its verified, if it finds a contradictory node within its path, thats a proof of absence, but if it cant find an individual node by its hash, then the proof is invalid. Once branches are seen as mini trees we can even do the following:
```

Comment:
```suggestion
There will be major changes to the format of `Branch`. I have realized that a `branch` is really just a sparse `tree`. Instead of using the array of nodes, I should be putting those nodes in the tree, then doing a regular `tree.get` to pull it back out. If the tree finds it, its verified, if it finds a contradictory node within its path, thats a proof of absence, but if it cant find an individual node by its hash, then the proof is invalid. Once branches are seen as mini trees we can even do the following:
```

Reply:
Capitalized letter in beginning of sentence. Super minor.